tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_power_iterative.h

```diff
@@ -12,24 +12,168 @@ using namespace sfpi;
 namespace ckernel {
 namespace sfpu {
 
+/**
+ * @brief Computes base raised to the power of pow (base**pow)
+ *
+ * This function implements binary exponentiation using a polynomial approximation algorithm
+ * based on "Simple Multiple Precision Algorithms for Exponential Functions [Tips & Tricks]"
+ * by Moroz et al. 2022 (https://doi.org/10.1109/MSP.2022.3157460).
+ * More specifically, it is the implementation of the `exp_21f` algorithm described in Section 5
+ *
+ * @param base The base value (sfpi::vFloat vector), can be any floating point number
+ * @param pow The exponent/power value (sfpi::vFloat vector), can be any floating point number
+ * @tparam IS_POSITIVE_EXPONENT If true, assumes exponent >= 0 (skips zero-base check for optimization)
+ *
+ * @return sfpi::vFloat Result of base**pow
+ *
+ * Special Cases:
+ * - base = 0, pow < 0: Returns NaN (undefined)
+ * - base < 0, pow = integer: Returns proper signed result (negative if odd power)
+ * - base < 0, pow = non-integer: Returns NaN (complex result)
+ * - Overflow/underflow: Clamped to appropriate limits
+ *
+ * @note This function assumes that the programmable constants are set to the following values:
+ * - vConstFloatPrgm0 = 1.4426950408889634f;
+ * - vConstFloatPrgm1 = -127.0f;
+ * - vConstFloatPrgm2 = std::numeric_limits<float>::quiet_NaN();
+ *
+ * @see Moroz et al. 2022 - "Simple Multiple Precision Algorithms for Exponential Functions"
+ *      ( https://doi.org/10.1109/MSP.2022.3157460 )
+ */
+template <bool IS_POSITIVE_EXPONENT>
+sfpi_inline sfpi::vFloat _sfpu_unary_power(sfpi::vFloat base, sfpi::vFloat pow) {
+    // The algorithm works in two steps:
+    // 1) Compute log2(base)
+    // 2) Compute base**pow = 2**(pow * log2(base))
+
+    // Step 1: Compute log2(base)
+    // Normalize base to calculation range
+    sfpi::vFloat abs_base = sfpi::abs(base);       // set base as positive
+    sfpi::vFloat x = sfpi::setexp(abs_base, 127);  // set exp to exp bias (put base in range of 1-2)
+
+    // 3rd order polynomial approx - determined using rminimax over [1,2]
+    sfpi::vFloat series_result = x * (x * (x * 0x2.44734p-4f - 0xd.e712ap-4f) + 0x2.4f5388p+0f) - 0x1.952992p+0f;
+
+    // Convert exponent to float
+    sfpi::vInt exp = sfpi::exexp(base);
+    v_if(exp < 0) { exp = sfpi::setsgn(~exp + 1, 1); }
+    v_endif;
+    sfpi::vFloat exp_f32 = sfpi::int32_to_float(exp, 0);
```

Comment:
Nitpick: Add comment to explain what this section does. It's not the only place where we have this but we should probably comment this appropriately because it is not obvious (naming of `int32_to_float` does not help).
 
```suggestion
    // sfpi::exexp returns signed-integer but sfpi::int32_to_float() takes sign-magnitude integers
    // These types differ on negative inputs, which is why we explicitly convert signed -> sign-magnitude
    v_if(exp < 0) { exp = sfpi::setsgn(~exp + 1, 1); }
    v_endif;
    sfpi::vFloat exp_f32 = sfpi::int32_to_float(exp, 0);
```